### PR TITLE
#3152 top navigation bar

### PR DESCRIPTION
**Pull request by AlexeyTuboltsev**
this PR replaces the old app layout with a new one. See #3152 for design details.

some details for the reviewers:
* Instead of having auth/unauth layouts there's only one layout component left. It selects the auth/unauth header automatically, devs don't have to care about it anymore. THe same applies to ErrorBoundary layouts.

* the `LayoutNew` is called this way to help the reviewers understand what has changed. I'll rename LayoutNew to simple `Layout` in this pr, after the review.

* menus and dropdown menus are generated from arrays of menu data (`MenuData.tsx`).
dropdown menu is built with a MenuBase component, otherwise it’s a normal menu

* here is a list of things that have been renamed and/or moved to other places:
-- WidgetGridLayout -> WidgetGrid
-- `ErrorBoundaryLayoutAuthorized` / `ErrorBoundaryLayoutUnuthorized`  -> `ErrorBoundaryLayout`
-- `LayoutAuthorized` /`LayoutUnauthorized` -> `LayoutNew` (mostly used as `withContainer(LayoutNew)`)

* there's an `InvestorIdentityButton.tsx` component that is hidden under a feature flag. I decided to completely remove it for now because the codebase has changed too much in the meantime and now it's non-functional and won't work even if we'd try to switch it on. Keeping it would add much complexity to this pr. If there's still need to have InvestorIdentity feature it should be done properly  from scratch via normal design process.

### CODE REVIEW

File: app/components/layouts/WidgetGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import * as cn from "classnames";
+import * as React from "react";
+
+import * as styles from "./WidgetGrid.module.scss";
+
+export const WidgetGrid: React.FunctionComponent<{
```
Comment (drakkein) [inline app/components/layouts/WidgetGrid.tsx]: We can use `CommonHtmlProps & TDataTestId` which will additionality have not used `style` property, but I will leave it up to you

File: app/components/layouts/header/Header.tsx [hunk 2]
```diff
@@ -2,127 +2,152 @@ import * as cn from "classnames";
...
 
-import { EUserType, TxPendingWithMetadata } from "../../../lib/api/users/interfaces";
+import { TxPendingWithMetadata } from "../../../lib/api/users/interfaces";
 import { actions } from "../../../modules/actions";
-import { selectIsAuthorized, selectUserType } from "../../../modules/auth/selectors";
 import { selectPlatformPendingTransaction } from "../../../modules/tx/monitor/selectors";
 import { appConnect } from "../../../store";
 import { appRoutes } from "../../appRoutes";
-import { Button, ButtonLink, EButtonLayout, EButtonTheme } from "../../shared/buttons";
+import { ButtonLink, EButtonTheme } from "../../shared/buttons/index";
```
Comment (drakkein) [inline app/components/layouts/header/Header.tsx]: we can shorten this import

File: app/components/layouts/header/Header.tsx [hunk 3]
```diff
@@ -2,127 +2,152 @@ import * as cn from "classnames";
...
 import { loginWalletRoutes, walletRegisterRoutes } from "../../wallet-selector/walletRoutes";
+import { MenuAuthorized } from "../menus/MenuAuthorized";
```
Comment (drakkein) [inline app/components/layouts/header/Header.tsx]: maybe we can add barrel module here, for all that menus imports?

File: app/components/layouts/header/PendingTransactionStatus.stories.tsx [hunk 4]
```diff
@@ -8,12 +8,14 @@ import { PendingTransactionStatus } from "./PendingTransactionStatus";
 storiesOf("PendingTransactionStatus", module)
   .add("No pending transactions", () => (
     <PendingTransactionStatus
+      className="className"
```
Comment (drakkein) [inline app/components/layouts/header/PendingTransactionStatus.stories.tsx]: Why do we add this className here?

File: app/components/layouts/menus/MenuEntry.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
+import * as cn from "classnames";
+import * as React from "react";
+import { match, NavLink } from "react-router-dom";
+
+import { TDataTestId, TTranslatedString } from "../../../types";
+import { invariant } from "../../../utils/invariant";
+import { InlineIcon } from "../../shared/icons/InlineIcon";
+import { ExternalLink } from "../../shared/links/index";
```
Comment (drakkein) [inline app/components/layouts/menus/MenuEntry.tsx]: we can shorten this import

File: app/components/layouts/menus/MenuEntry.tsx [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+type TMenuEntryExternalLink = { type: EMenuEntryType.EXTERNAL_LINK } & IMenuLink;
+
+type TMenuEntryLink = { type: EMenuEntryType.LINK } & IMenuLink;
+
+type TMenuEntryButton = { type: EMenuEntryType.ACTION } & IMenuAction;
+
+type TMenuEntrySeparator = { type: EMenuEntryType.SEPARATOR };
+
+export type TMenuEntry =
```
Comment (drakkein) [inline app/components/layouts/menus/MenuEntry.tsx]: This is kinda unreadable, can we separate these type groups to separate types, and then create union?

File: app/components/layouts/menus/MobileMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    closedElement={<img src={hamburger} alt="menu" />}
+    openElement={<img src={close} alt="menu" />}
+    className={styles.mobileMenu}
+    renderMenu={<Menu data={props.data} />}
+  />
+);
+
+export const MobileMenu = compose<IMenuData & IHandlers, {}>(
+  appConnect<IStateProps | null, IDispatchProps>({
+    stateToProps: state => {
```
Comment (drakkein) [inline app/components/layouts/menus/MobileMenu.tsx]: I'm not a fan of this construction, but TBH I'm not sure how we can improve it.

File: app/components/layouts/menus/MobileMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    isLinkActive: () => match => Boolean(match),
+  }),
+  withProps<IMenuData, IProps>(props => {
+    const accountMenu = accountMenuData(
+      props.actionRequired,
+      props.isLinkActive,
+      props.userType,
+      props.logout,
+    );
+    const data =
```
Comment (drakkein) [inline app/components/layouts/menus/MobileMenu.tsx]: Maybe we can move it to util function like `crateMenuData`?

File: app/components/dashboard/Dashboard.tsx [hunk 9]
```diff
@@ -43,4 +42,9 @@ export const Dashboard = compose<{}, {}>(
       d(actions.wallet.loadWalletData());
     },
   }),
+  withContainer(
+    withProps<{ "data-test-id"?: string }, {}>({ "data-test-id": "dashboard-application" })(
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply app/components/dashboard/Dashboard.tsx]: yep

File: app/components/layouts/Footer.tsx [hunk 10]
```diff
@@ -1,26 +1,24 @@
 import * as React from "react";
 import { FormattedMessage } from "react-intl-phraseapp";
 
-import { ExternalLink } from "../shared/links";
+import { ExternalLink } from "../shared/links/index";
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply app/components/layouts/Footer.tsx]: done

File: app/components/layouts/WidgetGrid.tsx (hunk 1 above)
Comment (AlexeyTuboltsev) [reply app/components/layouts/WidgetGrid.tsx]: fixed

File: app/components/layouts/header/Header.tsx (hunk 2 above)
Comment (AlexeyTuboltsev) [reply app/components/layouts/header/Header.tsx]: the problem with barrel imports is that you can't `Cmd + B` to the declaration which is super annoying. I'd rather leave it this way. 
Btw we discussed it once  it with Pawel and Moe, looks like nobody likes them and the general idea was to slowly get rid of them

File: app/components/layouts/header/Header.tsx (hunk 3 above)
Comment (AlexeyTuboltsev) [reply app/components/layouts/header/Header.tsx]: see above

File: app/components/layouts/header/PendingTransactionStatus.stories.tsx (hunk 4 above)
Comment (AlexeyTuboltsev) [reply app/components/layouts/header/PendingTransactionStatus.stories.tsx]: I need some css to properly place the button in the header

File: app/components/layouts/menus/MenuEntry.tsx (hunk 5 above)
Comment (AlexeyTuboltsev) [reply app/components/layouts/menus/MenuEntry.tsx]: see above

File: app/components/layouts/menus/MenuEntry.tsx (hunk 6 above)
Comment (AlexeyTuboltsev) [reply app/components/layouts/menus/MenuEntry.tsx]: you're right
done

File: app/components/layouts/menus/MobileMenu.tsx (hunk 7 above)
Comment (AlexeyTuboltsev) [reply app/components/layouts/menus/MobileMenu.tsx]: you mean `IStateProps | null` ? Not my favorite either, but there's not much we can do. This way at least the intention is clear

File: app/components/layouts/menus/MobileMenu.tsx (hunk 8 above)
Comment (AlexeyTuboltsev) [reply app/components/layouts/menus/MobileMenu.tsx]: well this function has so many parameters and needs so many imports, it's really a part of this `compose` thing, it's too tightly coupled to it. Don't know how to make it better

File: app/components/layouts/Footer.tsx [hunk 11]
```diff
@@ -1,26 +1,24 @@
...
-      <div className={styles.content}>
-        <img src={logo} className={styles.logo} alt="logo" />
-        <div>&copy; FIFTH FORCE GMBH</div>
-      </div>
-    </div>
+    <span className={styles.copyright}>©2019 Fifth Force GmbH</span>
+    <ExternalLink href="https://neufund.org/imprint" className={styles.link}>
+      <FormattedMessage id="footer.links.imprint" />
+    </ExternalLink>
+    <ExternalLink href="" className={styles.link}>
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply app/components/layouts/Footer.tsx]: I'll add the real one with a separate PR (there is one coming very soon with small visual fixes) For now I'd leave a todo here

**Pull request merged**